invenio_swh/schema.py

```diff
@@ -34,4 +38,25 @@ class Meta:
             "codeRepository",
             "programmingLanguage",
             "developmentStatus",
+            "deposit",
         )
+
+    def add_deposit_info(self, obj):
+        """Add deposit information to the codemeta."""
+        if hasattr(obj, "parent"):
+            parent_record = obj.parent
+        else:
+            parent_record = obj["parent"]
+
+        parent_doi = parent_record.get("pids", {}).get("doi")["identifier"]
+        origin_obj = {"swh:origin": {"@url": f"https://doi.org/{parent_doi}"}}
+
+        # If the record is the first version, it dumps `create_origin`
+        # Otherwise, dumps `add_to_origin`
+        is_first_version = obj.versions.index == 1
```

Comment:
question: is this meant for the first "version" (i.e. different DOIs) or revisions, i.e. for record metadata updates?